ragen/env/deepcoder/utils.py

```
@@ -0,0 +1,209 @@
+            if not ok:
+                return False, f"Runtime error on test {idx}: {err or out}"
+            got = out.strip()
+            exp = str(expected).strip()
+            if str(got) != str(exp):
+                return False, f"Wrong answer on test {idx}: got {got} expected {exp}"
+
+    return True, "All tests passed."
+
+
+def _exec_python(code: str, stdin_input: str, timeout_seconds: int = 5) -> Tuple[bool, str, str]:
+    debug_dir = "/tmp/deepcoder_debug"
```

Ruff (0.15.2)

[error] Probable insecure usage of temporary file or directory: "/tmp/deepcoder_debug"

(S108)